statemachine/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachine/CreateHandlerTest.java

```diff
@@ -142,12 +142,20 @@ public void testDefinitionReplacement() {
                 "        \"Lambda01\": \"${lambdaArn01}\"\n" +
                 "      }\n" +
                 "    }\n" +
+                "    \"number\": {\n" +
+                "      \"Resource\": ${number}\n" +
```

Comment:
Observation: we're inserting values into the map as strings instead of as numbers and booleans. The DefinitionSubstitutions property is still marked as a map of string to string, which is a bit unintuitive considering that we are providing support for non-string types. This typing also means we can't substitute an object (in the case that we want to substitute an entire state). Can we explore whether we can update the [resource schema](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-stepfunctions/blob/main/statemachine/aws-stepfunctions-statemachine.json#L105) to allow all types?

Reply:
I updated the resource schema in my latest commit!